rest-api-spec/src/main/resources/rest-api-spec/test/indices.stats/20_translog.yml

```diff
@@ -64,6 +64,52 @@ setup:
   - lte: { indices.test.primaries.translog.uncommitted_size_in_bytes: $creation_size }
   - match: { indices.test.primaries.translog.uncommitted_operations: 0 }
 
+---
+"Translog retention with soft_deletes":
+  - skip:
+      version: " - 7.9.99"
+      reason:  "start ignoring translog retention policy with soft-deletes enabled in 8.0"
+  - do:
+      indices.create:
+        index: test
+        body:
+          settings:
+            soft_deletes.enabled: true
+  - do:
+      cluster.health:
+        wait_for_no_initializing_shards: true
+  - do:
+      indices.stats:
+        metric: [ translog ]
+  - set: { indices.test.primaries.translog.size_in_bytes: creation_size }
+
+  - do:
+      index:
+        index: test
+        id:    1
+        body:  { "foo": "bar" }
+
+  - do:
+      indices.stats:
+        metric: [ translog ]
+  - gt: { indices.test.primaries.translog.size_in_bytes: $creation_size }
+  - match: { indices.test.primaries.translog.operations: 1 }
+  - match: { indices.test.primaries.translog.uncommitted_operations: 1 }
+  # call flush twice to sync the global checkpoint after the last operation so that we can have the safe commit
+  - do:
+      indices.flush:
+        index: test
+  - do:
+      indices.flush:
+        index: test
+  - do:
+      indices.stats:
+        metric: [ translog ]
+  - lt: { indices.test.primaries.translog.size_in_bytes: $creation_size }
```

Comment:
This is probably OK, but I am puzzled how the empty translog can be smaller than the initial translog (which I would assume is also empty)?

Reply:
After flushing, we have one empty translog file while an empty index has two: one when we create an empty translog https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/index/translog/Translog.java#L1878 and one when we open an engine. I added a comment.